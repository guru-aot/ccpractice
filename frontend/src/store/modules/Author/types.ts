export interface AuthorState {
  authorList: AuthorModel[];
  loading?: boolean;
  successStatus?: boolean;
  errorStatus?: boolean;
}
export interface AuthorModel {
  created: string;
  first_name: string;
  id: string;
  last_name: string;
  middle_name: string;
  modified: string;
}
