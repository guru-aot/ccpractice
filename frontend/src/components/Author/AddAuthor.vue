
/**
 * AuthorComponent
 */
 <template>
  <v-card class="mx-auto" style="max-width: 500px;">
    <v-alert type="success" v-if="successStatus">Author Added succesfully</v-alert>
    <v-alert type="error" v-if="errorStatus">Something went wrong...</v-alert>

    <v-toolbar color="deep-purple accent-4" cards dark flat>
      <v-card-title class="title font-weight-regular">Add Author</v-card-title>
      <v-spacer></v-spacer>
    </v-toolbar>
    <v-form ref="form" v-model="form" class="pa-4 pt-6">
      <v-text-field
        v-model="firstName"
        filled
        color="deep-purple"
        counter="10"
        label="First name"
        style="min-height: 96px"
        type="text"
        :rules="[rules.required]"
      ></v-text-field>
      <v-text-field
        v-model="lastName"
        filled
        color="deep-purple"
        counter="10"
        label="Last name"
        style="min-height: 96px"
        type="text"
        :rules="[rules.required]"
      ></v-text-field>
      <v-text-field
        v-model="middleName"
        filled
        color="deep-purple"
        counter="10"
        label="Middle name"
        style="min-height: 96px"
        type="text"
        :rules="[rules.required]"
      ></v-text-field>
    </v-form>
    <v-divider></v-divider>
    <v-card-actions>
      <v-btn text @click="$refs.form.reset()">Clear</v-btn>
      <v-spacer></v-spacer>
      <v-btn
        :disabled="!form"
        :loading="isLoading"
        class="white--text"
        color="deep-purple accent-4"
        depressed
        @click="addAuthor"
      >Submit</v-btn>
    </v-card-actions>
  </v-card>
</template>
 <script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
const AuthorModule = namespace('AuthorModule');

@Component
export default class AuthorComponent extends Vue {
  @AuthorModule.Getter('successStatus') public successStatus!: boolean;
  @AuthorModule.Getter('errorStatus') public errorStatus!: boolean;
  @AuthorModule.Action('addAuthor') public addAuthorStore!: any;
  public form: boolean = false;
  private isLoading: boolean = false;
  private firstName: string = '';
  private lastName: string = '';
  private middleName: string = '';

  private rules = {
    length: (len: any) => (v: any) =>
      (v || '').length >= len || 'Invalid character length, required' + len,

    required: (v: any) => !!v || 'This field is required'
  };
  /**
   * @description
   */
  private addAuthor() {
    const data = {
      first_name: this.firstName,
      last_name: this.lastName,
      middle_name: this.middleName
    };

    this.addAuthorStore(data);
    // this.$refs.form.reset();
    (this.$refs.form as HTMLFormElement).reset();
  }
}
</script>
 