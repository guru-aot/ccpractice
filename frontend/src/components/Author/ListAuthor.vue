

  /**
   * ListViewComponent Page
   */
  
  <template>
  <v-card class="mx-auto">
    <v-toolbar flat>
      <v-toolbar-title class="grey--text">All Author</v-toolbar-title>
      <div class="flex-grow-1"></div>
    </v-toolbar>
    <v-divider></v-divider>
    <v-card-text v-if="getAuthorList && getAuthorList.length == 0">No Author found. Add one</v-card-text>
    <v-card-text v-if="getAuthorList && getAuthorList.length != 0">
      <v-simple-table fixed-header>
        <template v-slot:default>
          <thead>
            <tr>
              <th class="text-left">SI</th>
              <th class="text-left">first_name</th>
              <th class="text-left">last_name</th>
              <th class="text-left">middle_name</th>
            </tr>
          </thead>
          <tbody>
            <tr v-for="(author ,idx) in  getAuthorList" :key="author.id">
              <td>{{ idx+1 }}</td>
              <td>{{ author.first_name }}</td>
              <td>{{ author.last_name }}</td>
              <td>{{ author.middle_name }}</td>
            </tr>
          </tbody>
        </template>
      </v-simple-table>
    </v-card-text>
  </v-card>
</template>
  <script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
const AuthorModule = namespace('AuthorModule');

@Component
export default class ListAuthorComponent extends Vue {
  @AuthorModule.Action('loadAuthor') public loadAuthor!: any;
  @AuthorModule.Getter('getAuthorList') public getAuthorList!: [];

  private mounted() {
    this.loadAuthor();
  }
}
</script>
 