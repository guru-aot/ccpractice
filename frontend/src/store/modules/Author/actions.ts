import { ActionTree } from 'vuex';
import { AuthorState, AuthorModel } from './types';
import { RootState } from '../../types';
import { BASE_URL, AUTHORURL } from '@/config/urlList';
import axios from '@/lib/axios';

/**
 * author Actions
 *
 */
export const actions: ActionTree<AuthorState, RootState> = {
  /**
   * @param  {} {commit, dispatch}
   * @param  {} author  list
   */
  addAuthor({ commit, dispatch }, data) {
    commit('SET_LOADING', true);
    axios
      .post(BASE_URL + AUTHORURL, data)
      .then(_ => {
        commit('SET_LOADING', false);
        commit('SET_AUTHOR_SUCCESSFULLY', true);
        commit('SET_AUTHOR_ERROR', false);
        dispatch('loadAuthor');
      })
      .catch(() => {
        commit('SET_CONTACT_SUCCESSFULLY', false);
        commit('SET_CONTACT_ERROR', true);
      });
  },

  /**
   * load author from server and set to store
   * @param {*} { commit }
   */
  loadAuthor({ commit }) {
    commit('SET_LOADING', true);
    axios
      .get(BASE_URL + AUTHORURL)
      .then((r: any) => r.data)
      .then((data: AuthorModel[]) => {
        commit('SET_AUTHORLIST', data);
        commit('SET_LOADING', false);
      });
  },
  /**
   * clear message
   * @param {*} { commit }
   */
  clearStatus({ commit }) {
    commit('SET_AUTHOR_SUCCESSFULLY', false);
    commit('SET_AUTHOR_ERROR', false);
  },
};
