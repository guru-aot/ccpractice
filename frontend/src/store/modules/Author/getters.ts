import { GetterTree } from 'vuex';
import { AuthorState } from './types';
import { RootState } from '../../types';

/**
 * Getters
 */
export const getters: GetterTree<AuthorState, RootState> = {
  /**
   * author
   * @param {*} state
   * @returns  {object} author
   */

  getAuthorList(state) {
    return state.authorList;
  },
  successStatus(state) {
    return state.successStatus;
  },
  errorStatus(state) {
    return state.errorStatus;
  },
};
