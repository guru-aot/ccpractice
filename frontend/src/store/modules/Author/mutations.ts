import { MutationTree } from 'vuex';
import { AuthorState } from './types';

/**
 * mutation
 */
export const mutations: MutationTree<AuthorState> = {
  /**
   * Sets loading
   * @param {*} state
   * @param {boolean} flag - loading
   */

  SET_LOADING(state, flag: boolean) {
    state.loading = flag;
  },
  /**
   * Sets author
   * @param {*} state
   * @param payload author list
   */

  SET_AUTHORLIST(state, payload: any) {
    state.authorList = payload;
  },

  /**
   * set stae on success
   * @param {*} state
   * @param {*} payload
   */
  SET_AUTHOR_SUCCESSFULLY(state, payload: any) {
    state.successStatus = payload;
  },

  /**
   * set stae on error
   * @param {*} state
   * @param {*} payload
   */
  SET_AUTHOR_ERROR(state, payload: any) {
    state.errorStatus = payload;
  },
};
