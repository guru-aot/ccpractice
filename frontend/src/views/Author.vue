<template>
  <div>
    <div>
      <AddAuthor />
    </div>
    <div class="mt-6">
      <ListAuthor />
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import AddAuthor from '@/components/Author/AddAuthor.vue'; // @ is an alias to /src
import ListAuthor from '@/components/Author/ListAuthor.vue';
@Component({
  components: {
    AddAuthor,
    ListAuthor
  }
})
export default class Author extends Vue {}
</script>
