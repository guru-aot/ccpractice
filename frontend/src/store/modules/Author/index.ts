import { Module } from 'vuex';
import { getters } from './getters';
import { actions } from './actions';
import { mutations } from './mutations';
import { AuthorState } from './types';
import { RootState } from '../../types';

/**
 * author state
 */
export const state: AuthorState = {
  authorList: [],
  loading: false,
  successStatus: false,
  errorStatus: false,
};

const namespaced: boolean = true;

const AuthorModule: Module<AuthorState, RootState> = {
  namespaced,
  state,
  getters,
  actions,
  mutations,
};

export default AuthorModule;
