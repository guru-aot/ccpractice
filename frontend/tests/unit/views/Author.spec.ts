import { shallowMount, createLocalVue } from '@vue/test-utils';
import Author from '@/views/Author.vue';
import Vuex from 'vuex';
import vuetify from 'vuetify';

describe('Author.vue', () => {
  let wrapper: any;
  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(Vuex);
    localVue.use(vuetify);

    const store = new Vuex.Store({
      modules: {
        AuthorModule: {
          state: { authors: [] },
          getters: {
            getAuthorList: jest.fn(),
          },
        },
      },
    });
    wrapper = shallowMount(Author, {
      localVue,
      store,
    });
  });
  it('renders without crashing', () => {
    // const wrapper = shallowMount(Author);
    expect(wrapper.element).toMatchSnapshot();
  });
});
