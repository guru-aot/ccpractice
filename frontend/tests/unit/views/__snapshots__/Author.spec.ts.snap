// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author.vue renders without crashing 1`] = `
<div>
  <div>
    <addauthor-stub />
  </div>
   
  <div
    class="mt-6"
  >
    <listauthor-stub />
  </div>
</div>
`;
